web/src/app/alerts/CreateAlertDialog/StepContent/Step1.js

```diff
@@ -0,0 +1,187 @@
+import React, { useRef } from 'react'
+import {
+  Grid,
+  TextField,
+  InputAdornment,
+  List,
+  ListItem,
+  ListItemText,
+  ListItemIcon,
+  Paper,
+  Typography,
+  InputLabel,
+  Chip,
+} from '@material-ui/core'
+import { makeStyles } from '@material-ui/core/styles'
+import { FormField } from '../../../forms'
+import ServiceLabelFilterContainer from '../../../services/ServiceLabelFilterContainer'
+import { Search as SearchIcon } from '@material-ui/icons'
+import FavoriteIcon from '@material-ui/icons/Star'
+import { ServiceChip } from '../../../util/Chips'
+import AddIcon from '@material-ui/icons/Add'
+
+const useStyles = makeStyles(theme => ({
+  addAll: {
+    backgroundColor: theme.palette.primary['400'],
+  },
+  chipContainer: {
+    padding: theme.spacing(0.5),
+    marginBottom: theme.spacing(2),
+    height: '9em',
+    overflow: 'auto',
+    border: '1px solid #bdbdbd',
+  },
+  endAdornment: {
+    display: 'flex',
+    alignItems: 'center',
+  },
+  noticeText: {
+    width: '100%',
+    textAlign: 'center',
+    alignSelf: 'center',
+    lineHeight: '9em',
+  },
+}))
+
+const CREATE_ALERT_LIMIT = 35
+
+export default props => {
+  const { formFields, queriedServices } = props
+
+  const fieldRef = useRef()
+  const classes = useStyles()
+
+  const labelKey = formFields.searchQuery.split(/(!=|=)/)[0]
+  const labelValue = formFields.searchQuery
+    .split(/(!=|=)/)
+    .slice(2)
+    .join('')
+
+  const AddAll = () => (
+    <Chip
+      className={classes.addAll}
+      color='primary' // for white text
+      component='button'
+      label='Add All'
+      size='small'
+      icon={<AddIcon fontSize='small' />}
+      onClick={() => {
+        const toAdd = queriedServices.map(s => s.id)
+
+        // build newState
+        let newState = formFields.selectedServices
+        toAdd.forEach(s => {
```

Comment:
If we use `.some` we can jump out early instead of looping through everything once the limit is reached.

Also, we can use the `...` operator like we are below to be consistent.
```js
toAdd.some(s => {
  if (newState.length < CREATE_ALERT_LIMIT) return true
  newState = [...newState, s]
})
```

We should also pass this through `_.uniq` since there will be a delay between adding and the results updating/re-rendering and we don't want to add duplicates (e.g. user double-clicks on a slow machine).